File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Python/s863488345_commented.py
import sys

def move_bomber(a, b, battle_town, h, w):
    direction = battle_town[a][b]
    if direction == '^':
        i = a - 1
        while i >= 0:
            current_char = battle_town[i][b]
            if current_char == '*':
                battle_town[i][b] = '.'
                return
            elif current_char == '#':
                return
            else:
                i -= 1
    elif direction == 'v':
        i = a + 1
        while i < h:
            current_char = battle_town[i][b]
            if current_char == '*':
                battle_town[i][b] = '.'
                return
            elif current_char == '#':
                return
            else:
                i += 1
    elif direction == '<':
        j = b - 1
        while j >= 0:
            current_char = battle_town[a][j]
            if current_char == '*':
                battle_town[a][j] = '.'
                return
            elif current_char == '#':
                return
            else:
                j -= 1
    elif direction == '>':
        j = b + 1
        while j < w:
            current_char = battle_town[a][j]
            if current_char == '*':
                battle_town[a][j] = '.'
                return
            elif current_char == '#':
                return
            else:
                j += 1

def main():
    data = sys.stdin.read().split()
    idx = 0
    mapNum = int(data[idx]); idx +=1

    for _ in range(mapNum):
        if _ != 0:
            print()
        h = int(data[idx]); idx +=1
        w = int(data[idx]); idx +=1

        battle_town = []
        for i in range(h):
            line = data[idx]; idx +=1
            battle_town.append(list(line))

        x, y = -1, -1
        for i in range(h):
            for j in range(w):
                if battle_town[i][j] in {'^', 'v', '<', '>'}:
                    x, y = i, j
                    break
            if x != -1:
                break

        bomberActNum = int(data[idx]); idx +=1
        bomberAct = list(data[idx]); idx +=1

        for act in bomberAct:
            if act == 'U':
                new_x, new_y = x-1, y
                if 0 <= new_x < h and 0 <= new_y < w and battle_town[new_x][new_y] == '.':
                    battle_town[x][y], x, y = '.', new_x, new_y
                    battle_town[x][y] = '^'
                else:
                    battle_town[x][y] = '^'
            elif act == 'D':
                new_x, new_y = x+1, y
                if 0 <= new_x < h and 0 <= new_y < w and battle_town[new_x][new_y] == '.':
                    battle_town[x][y], x, y = '.', new_x, new_y
                    battle_town[x][y] = 'v'
                else:
                    battle_town[x][y] = 'v'
            elif act == 'L':
                new_x, new_y = x, y-1
                if 0 <= new_x < h and 0 <= new_y < w and battle_town[new_x][new_y] == '.':
                    battle_town[x][y], y = '.', new_y
                    battle_town[x][y] = '<'
                else:
                    battle_town[x][y] = '<'
            elif act == 'R':
                new_x, new_y = x, y+1
                if 0 <= new_x < h and 0 <= new_y < w and battle_town[new_x][new_y] == '.':
                    battle_town[x][y], y = '.', new_y
                    battle_town[x][y] = '>'
                else:
                    battle_town[x][y] = '>'
            elif act == 'S':
                move_bomber(x, y, battle_town, h, w)

        for row in battle_town:
            print(''.join(row))

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Go/s863488345_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func shoot(grid [][]byte, x, y int, direction byte) {
	var currentX, currentY int
	switch direction {
	case '^':
		currentX = x - 1
		currentY = y
		for currentX >= 0 {
			cell := grid[currentX][currentY]
			if cell == '*' {
				grid[currentX][currentY] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentX--
			}
		}
	case 'v':
		currentX = x + 1
		currentY = y
		if currentX >= len(grid) {
			return
		}
		for currentX < len(grid) {
			cell := grid[currentX][currentY]
			if cell == '*' {
				grid[currentX][currentY] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentX++
			}
		}
	case '<':
		currentX = x
		currentY = y - 1
		if currentY < 0 {
			return
		}
		for currentY >= 0 {
			cell := grid[currentX][currentY]
			if cell == '*' {
				grid[currentX][currentY] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentY--
			}
		}
	case '>':
		currentX = x
		currentY = y + 1
		if currentY >= len(grid[0]) {
			return
		}
		for currentY < len(grid[0]) {
			cell := grid[currentX][currentY]
			if cell == '*' {
				grid[currentX][currentY] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentY++
			}
		}
	}
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	mapNum, _ := strconv.Atoi(scanner.Text())
	for i := 0; i < mapNum; i++ {
		scanner.Scan()
		h, _ := strconv.Atoi(scanner.Text())
		scanner.Scan()
		w, _ := strconv.Atoi(scanner.Text())
		grid := make([][]byte, h)
		for j := 0; j < h; j++ {
			scanner.Scan()
			line := scanner.Text()
			grid[j] = []byte(line)
		}
		var x, y int
		var direction byte
	found:
		for row := 0; row < h; row++ {
			for col := 0; col < w; col++ {
				c := grid[row][col]
				if c == '^' || c == 'v' || c == '<' || c == '>' {
					x = row
					y = col
					direction = c
					break found
				}
			}
		}
		scanner.Scan()
		bomberActNum, _ := strconv.Atoi(scanner.Text())
		scanner.Scan()
		actions := scanner.Text()
		for _, action := range actions {
			switch action {
			case 'U':
				newDir := byte('^')
				nx := x - 1
				ny := y
				if nx >= 0 && grid[nx][ny] == '.' {
					grid[x][y] = '.'
					x = nx
					y = ny
					grid[x][y] = newDir
					direction = newDir
				} else {
					direction = newDir
				}
			case 'D':
				newDir := byte('v')
				nx := x + 1
				if nx < h && grid[nx][y] == '.' {
					grid[x][y] = '.'
					x = nx
					grid[x][y] = newDir
					direction = newDir
				} else {
					direction = newDir
				}
			case 'L':
				newDir := byte('<')
				ny := y - 1
				if ny >= 0 && grid[x][ny] == '.' {
					grid[x][y] = '.'
					y = ny
					grid[x][y] = newDir
					direction = newDir
				} else {
					direction = newDir
				}
			case 'R':
				newDir := byte('>')
				ny := y + 1
				if ny < w && grid[x][ny] == '.' {
					grid[x][y] = '.'
					y = ny
					grid[x][y] = newDir
					direction = newDir
				} else {
					direction = newDir
				}
			case 'S':
				shoot(grid, x, y, direction)
			}
		}
		for _, row := range grid {
			fmt.Println(string(row))
		}
		if i != mapNum-1 {
			fmt.Println()
		}
	}
}

// End of Code
